src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -237,20 +236,26 @@ public bool TryAddModelError(string key, Exception exception, ModelMetadata meta
             if (exception is FormatException || exception is OverflowException)
             {
                 // Convert FormatExceptions and OverflowExceptions to Invalid value messages.
-                ModelStateEntry entry;
-                TryGetValue(key, out entry);
+                TryGetValue(key, out var entry);
 
-                var name = metadata.GetDisplayName();
+                var messageProvider = metadata.ModelBindingMessageProvider;
+                var name = metadata.DisplayName ?? metadata.PropertyName;
                 string errorMessage;
-                if (entry == null)
+                if (entry == null && name == null)
```

Comment:
I'd suggest documenting something like *the whole reason why we don't always have a name is that we don't want to use `GetDisplayName()` because reasons...*

What's a comment you can add that would prevent future us from *simplifying* this code.